google/cloud/aiplatform/metadata/schema/utils.py

```diff
@@ -143,6 +143,154 @@ def to_dict(self):
         return results
 
 
+@dataclass
+class AnnotationSpec:
+    """A class that represents the annotation spec of a Confusion Matrix.
+    Args:
+        display_name (str):
+            Optional. Display name for a column of a confusion matrix.
+        id (str):
+            Optional. Id for a column of a confusion matrix.
+    """
+
+    display_name: Optional[str] = None
+    id: Optional[str] = None
+
+    def to_dict(self):
+        """ML metadata schema dictionary representation of this DataClass"""
+        results = {}
+        if self.display_name:
+            results["displayName"] = self.display_name
+        if self.id:
+            results["id"] = self.id
+
+        return results
+
+
+@dataclass
+class ConfusionMatrix:
+    """A class that represents a Confusion Matrix.
+    Args:
+        matrix (List[List[int]]):
+            Required. A 2D array of integers that represets the values for the confusion matrix.
+        annotation_specs: (List(AnnotationSpec)):
+            Optional. List of column annotation specs which contains display_name (str) and id (str)
+    """
+
+    matrix: List[List[int]]
+    annotation_specs: Optional[List[AnnotationSpec]] = None
+
+    def to_dict(self):
+        """ML metadata schema dictionary representation of this DataClass"""
+        results = {}
+        if self.annotation_specs:
```

Comment:
We may want to add a validation to check the number of columns specified in annotation specs matches the the confusion matrix dimensions. If so please add a negative test for it in the unit tests as well.